File: src/utils/validators.ts
import { ValidationError } from '../types/errors';

// Validar que un valor no sea nulo o vacío
export function validateRequired(value: any, fieldName: string): void {
  if (value === null || value === undefined || value === '') {
    throw new ValidationError(`${fieldName} is required`);
  }
}

// Validar que un número sea positivo
export function validatePositiveNumber(value: number, fieldName: string): void {
  if (typeof value !== 'number' || value <= 0) {
    throw new ValidationError(`${fieldName} must be a positive number`);
  }
}

// Validar que sea una URL válida
export function validateUrl(url: string, fieldName: string): void {
  try {
    new URL(url);
  } catch {
    throw new ValidationError(`${fieldName} must be a valid URL`);
  }
}

// Validar email
export function validateEmail(email: string): boolean {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
}

// Validar RUT chileno (formato básico)
export function validateRut(rut: string): boolean {
  const rutRegex = /^[0-9]{1,2}\.[0-9]{3}\.[0-9]{3}-[0-9kK]$/;
  return rutRegex.test(rut);
}

File: kotlin/src/main/kotlin/cl/payments/utils/Validators.kt
package cl.payments.utils

import cl.payments.types.ValidationException
import java.net.URL

// Validar que un valor no sea nulo o vacío
fun validateRequired(value: String?, fieldName: String) {
    if (value.isNullOrBlank()) {
        throw ValidationException("$fieldName is required")
    }
}

// Validar que un número sea positivo
fun validatePositiveNumber(value: Int, fieldName: String) {
    if (value <= 0) {
        throw ValidationException("$fieldName must be a positive number")
    }
}

// Validar que sea una URL válida
fun validateUrl(url: String, fieldName: String) {
    try {
        URL(url)
    } catch (e: Exception) {
        throw ValidationException("$fieldName must be a valid URL")
    }
}

// Validar email
fun validateEmail(email: String): Boolean {
    val emailRegex = "^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$".toRegex()
    return emailRegex.matches(email)
}

// Validar RUT chileno (formato básico)
fun validateRut(rut: String): Boolean {
    val rutRegex = "^[0-9]{1,2}\\.[0-9]{3}\\.[0-9]{3}-[0-9kK]$".toRegex()
    return rutRegex.matches(rut)
}